src/main/java/org/spongepowered/common/service/scheduler/AsyncScheduler.java

```diff
@@ -106,608 +51,139 @@
     private final Condition condition = this.lock.newCondition();
     // The dynamic thread pooling executor of asynchronous tasks.
     private ExecutorService executor;
-    // Query actor for task information
-    private SchedulerHelper schedulerHelper;
 
     private AsyncScheduler() {
```

Comment:
It will make no difference to plugins whether it's a singleton or not and I don't see a problem with it being one. There's no use-case for more than one instance from what I can see.